langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AzureAiSearchContentRetriever.java

```diff
@@ -0,0 +1,348 @@
+         * @param tokenCredential the credentials to authenticate with Azure Active Directory
+         * @return builder
+         */
+        public AzureAiSearchContentRetrieverBuilder tokenCredential(TokenCredential tokenCredential) {
+            this.tokenCredential = tokenCredential;
+            return this;
+        }
+
+        /**
+         * If using the ready-made index, sets the number of dimensions of the embeddings.
+         * This parameter is exclusive of the index parameter.
+         *
+         * @param dimensions The number of dimensions of the embeddings.
+         * @return builder
+         */
+        public AzureAiSearchContentRetrieverBuilder dimensions(int dimensions) {
+            this.dimensions = dimensions;
+            return this;
+        }
+
+        /**
+         * If using a custom index, sets the index to be used.
+         * This parameter is exclusive of the dimensions parameter.
+         *
+         * @param index The index to be used.
+         * @return builder
+         */
+        public AzureAiSearchContentRetrieverBuilder index(SearchIndex index) {
+            this.index = index;
+            return this;
+        }
+
+        /**
+         * Sets the Embedding Model.
+         *
+         * @param embeddingModel The Embedding Model.
+         * @return builder
+         */
+        public AzureAiSearchContentRetrieverBuilder embeddingModel(EmbeddingModel embeddingModel) {
+            this.embeddingModel = embeddingModel;
+            return this;
+        }
+
+        /**
+         * Sets the maximum number of {@link Content}s to retrieve.
+         *
+         * @param maxResults The maximum number of {@link Content}s to retrieve.
+         * @return builder
+         */
+        public AzureAiSearchContentRetrieverBuilder maxResults(int maxResults) {
+            this.maxResults = maxResults;
+            return this;
+        }
+
+        /**
+         * Sets the minimum relevance score for the returned {@link Content}s.
+         * {@link Content}s scoring below {@code #minScore} are excluded from the results.
+         *
+         * @param minScore The minimum relevance score for the returned {@link Content}s.
+         * @return builder
+         */
+        public AzureAiSearchContentRetrieverBuilder minScore(double minScore) {
+            this.minScore = minScore;
+            return this;
+        }
+
+        /**
+         * Sets the Azure AI Search Query Type.
+         *
+         * @param azureAiSearchQueryType The Azure AI Search Query Type.
+         * @return builder
+         */
+        public AzureAiSearchContentRetrieverBuilder queryType(AzureAiSearchQueryType azureAiSearchQueryType) {
+            this.azureAiSearchQueryType = azureAiSearchQueryType;
+            return this;
+        }
+
+        public AzureAiSearchContentRetriever build() {
+            ensureNotNull(endpoint, "endpoint");
```

Comment:
I would move all these checks into a single constructor that accepts all fields and validate it there.
And other constructors can invoke this "full" constructor.
Otherwise, users of constructors do not pass these checks.

Reply:
Indeed! This is done